src/Razor/src/Microsoft.NET.Sdk.Razor/build/netstandard2.0/Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets

```diff
@@ -0,0 +1,103 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.Sdk.Razor.ApplicationPartsDiscovery
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+
+<Project ToolsVersion="14.0">
+  <UsingTask
+    TaskName="Microsoft.AspNetCore.Razor.Tasks.DiscoverMvcApplicationParts"
+    AssemblyFile="$(RazorSdkBuildTasksAssembly)"
+    Condition="'$(RazorSdkBuildTasksAssembly)' != ''" />
+
+  <PropertyGroup>
+    <GenerateMvcApplicationPartsAttribute Condition="'$(GenerateMvcApplicationPartsAssemblyAttributes)' == '' AND '$(OutputType)' == 'Exe'">true</GenerateMvcApplicationPartsAttribute>
+
+    <CoreCompileDependsOn Condition="'$(GenerateMvcApplicationPartsAttribute)' == 'true' AND '$(DesignTimeBuild)' != 'true'">
+      _DiscoverMvcApplicationParts;
+      $(CoreCompileDependsOn);
+    </CoreCompileDependsOn>
+
+    <_MvcApplicationPartAttributeGeneratedFile>$(IntermediateOutputPath)$(TargetName).MvcApplicationPartsAssemblyInfo$(DefaultLanguageSourceExtension)</_MvcApplicationPartAttributeGeneratedFile>
+
+    <_DiscoverMvcApplicationPartsDependsOn>
+      ResolveAssemblyReferences;
+      _InitializeDiscoverMvcApplicationParts;
+    </_DiscoverMvcApplicationPartsDependsOn>
+  </PropertyGroup>
+
+  <Target Name="_InitializeDiscoverMvcApplicationParts">
+    <!-- Can be used by testassets which need to simulate a reference to Microsoft.AspNetCore.App -->
+    <PropertyGroup Condition="'$(_ReferencesMicrosoftAspNetCoreApp)' == ''">
+      <_ReferencesMicrosoftAspNetCoreApp>@(FrameworkReference->AnyHasMetadataValue('Identity', 'Microsoft.AspNetCore.App'))</_ReferencesMicrosoftAspNetCoreApp>
+    </PropertyGroup>
+  </Target>
```

Comment:
I'd like to understand more about this? What kinds of test can't we write without it?

I'm also not sure that this works. 

IIRC The dependencies of `_DiscoverMvcApplicationParts` won't be executed unless `_DiscoverMvcApplicationParts` is going to be. So using the dependencies of a target to configure whether the target runs doesn't super work.

Reply:
It's possible to run a target that determines whether a later target will run, but it is complex and subtle. If you can avoid that situation, please do so. If you can't, loop me back in and I'll look at it with a magnifying glass.

Reply:
This was obviously incorrect, but I think the scenario it's attempting to rule out is a vanishingly small number or projects (targets Razor or Web SDK, targets `netcoreapp3.0` or later, but does not reference Microsoft.AspNetCore.App. Plus in the most ordinary case, it'll end up ruling that there's nothing to do here.